tests/ignite/metrics/test_accumulation.py

```diff
@@ -129,44 +133,36 @@ def test_geom_average():
     np.testing.assert_almost_equal(m.numpy(), _geom_mean(y_true.reshape(-1, 10)), decimal=5)
 
 
-def test_integration():
-    def _test(metric_cls, true_result_fn):
+@pytest.mark.parametrize("metric_cls, true_result_fn", [(Average, _mean), (GeometricAverage, _geom_mean)])
+def test_integration(metric_cls, true_result_fn):
 
-        size = 100
-        custom_variable = 10.0 + 5.0 * torch.rand(size, 12)
+    size = 100
+    custom_variable = 10.0 + 5.0 * torch.rand(size, 12)
 
-        def update_fn(engine, batch):
-            return 0, custom_variable[engine.state.iteration - 1]
-
-        engine = Engine(update_fn)
-
-        custom_var_mean = metric_cls(output_transform=lambda output: output[1])
-        custom_var_mean.attach(engine, "agg_custom_var")
+    def update_fn(engine, batch):
+        return 0, custom_variable[engine.state.iteration - 1]
 
-        state = engine.run([0] * size)
-        np.testing.assert_almost_equal(
-            state.metrics["agg_custom_var"].numpy(), true_result_fn(custom_variable), decimal=5
-        )
+    engine = Engine(update_fn)
 
-        size = 100
-        custom_variable = 10.0 + 5.0 * torch.rand(size)
+    custom_var_mean = metric_cls(output_transform=lambda output: output[1])
+    custom_var_mean.attach(engine, "agg_custom_var")
 
-        def update_fn(engine, batch):
-            return 0, custom_variable[engine.state.iteration - 1].item()
+    state = engine.run([0] * size)
+    np.testing.assert_almost_equal(state.metrics["agg_custom_var"].numpy(), true_result_fn(custom_variable), decimal=5)
 
-        engine = Engine(update_fn)
+    size = 100
+    custom_variable = 10.0 + 5.0 * torch.rand(size)
 
-        custom_var_mean = metric_cls(output_transform=lambda output: output[1])
-        custom_var_mean.attach(engine, "agg_custom_var")
+    def update_fn(engine, batch):
+        return 0, custom_variable[engine.state.iteration - 1].item()
```

Comment:
Here you can conditionally apply `.item()` if `custom_variable[engine.state.iteration - 1]` is a scalar

Reply:
Sure but won't this always be a scalar? Since we only defined the test?

Reply:
I think no as there are 2 types of input we test on

Reply:
Oh so you want me to handle both cases with one function

Reply:
Yes, I want to handle input size as a parameter, https://github.com/pytorch/ignite/pull/2620#discussion_r924135827

Reply:
The point is to reduce the number of lines of code. We have 4 lines of duplicated code: 
```python
    size = 100  # duplicated
    custom_variable = 10.0 + 5.0 * torch.rand(size)

    def update_fn(engine, batch):
        return 0, custom_variable[engine.state.iteration - 1].item()

    engine = Engine(update_fn)  # duplicated

    custom_var_mean = metric_cls(output_transform=lambda output: output[1])  # duplicated
    custom_var_mean.attach(engine, "agg_custom_var")  # duplicated

    state = engine.run([0] * size)  # duplicated
    assert state.metrics["agg_custom_var"] == pytest.approx(true_result_fn(custom_variable))
```

Reply:
Also one more thing. device is defied separately as in every function 
![image](https://user-images.githubusercontent.com/43926001/179753066-b57d57e0-68ab-49b5-9faf-83e544c81909.png)
except
![image](https://user-images.githubusercontent.com/43926001/179752974-ffc8c760-8768-439e-900b-d425813b3322.png).

We have already put a check in this function to execute this only when we have GPU. So can I just declare device as `device = idist.device()` globally? This will help us parameterize a lot of functions.

Reply:
Test functions where we define `device = idist.device()` are specific for distributed tests. We can talk on discord about them and how to improve these tests.
I do not think we can define `device` globally as we may alter test setup...